account_move_tree_purchase_date_origin/models/account_move.py

```diff
@@ -0,0 +1,21 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import fields, models, api
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    purchase_date_origin = fields.Datetime(compute="_compute_purchase_date_origin")
+
+    @api.depends('invoice_line_ids')
+    def _compute_purchase_date_origin(self):
+        for record in self:
+            purchases = record.invoice_line_ids.mapped("purchase_line_id.order_id")
+            if len(purchases) > 1:
+                record.purchase_date_origin = min(purchases.mapped("date_order"))
+            elif len(purchases) == 1:
+                record.purchase_date_origin = purchases.date_order
+            else:
+                record.purchase_date_origin = False
```

Comment:
Equivalent code (not tested):
```suggestion
            record.purchase_date_origin = (
                purchases
                and purchases.mapped("date_order").sort(key=lambda x: x.date_order)[0]
                or False
            )
```